growth/too/models.py

```diff
@@ -728,6 +728,79 @@ def flat(self):
         else:
             return self.flat_2d,
 
+    def observation_ipix(self, telescope, filt, start_time, end_time):
+        
+        observation_list = self.get_observations(filt, start_time, end_time)
+        field_ipix = {observation.field.ipix for observation in observation_list
+                        if observation.field.ipix is not None}
+        return field_ipix
+
+    def observation_area(self, telescope, filt, start_time, end_time):
+        nside = self.nside
+        ipix = self.observation_ipix(filt, start_time, end_time)
+        return hp.nside2pixarea(nside, degrees=True) * len(ipix)
+
+    def observation_nexp(self, telescope, filt, start_time, end_time):
+        observation_list = self.get_observations(filt, start_time, end_time)
+        return len({observation.observation_id for observation in observation_list})
+
+    def observation_starttime(self, telescope, filt, start_time, end_time):
+        observation_list = self.get_observations(filt, start_time, end_time)
+        if len(observation_list) == 0:
+            return None
+        return min(observation.obstime for observation in observation_list)
+
+    def observation_totaltime(self, telescope, filt, start_time, end_time):
+        observation_list = self.get_observations(filt, start_time, end_time)
+        tot = 0
+        observation_ids = []
+        for ii, observation in enumerate(observation_list):
```

Comment:
Please rewrite this as a comprehension.

Reply:
I couldn't figure out how to do this without having two list comprehensions instead of the for loop/appending.